Execution fails

Hi,

Execution runs in Eclipse but in CLI it fails and I can't figure out why.
This is the error I got:

Feb 09, 2021 9:20:02 PM opennlp.uima.chunker.Chunker initialize(113)
INFO: Initializing the OpenNLP Chunker annotator.
org.apache.uima.resource.ResourceAccessException
	at org.apache.uima.impl.UimaContext_ImplBase.getResourceFilePath(UimaContext_ImplBase.java:359)
	at org.ohnlp.medxn.ae.ACLookupDrugNormAnnotator.initialize(ACLookupDrugNormAnnotator.java:65)
	at org.apache.uima.analysis_engine.impl.PrimitiveAnalysisEngine_impl.initializeAnalysisComponent(PrimitiveAnalysisEngine_impl.java:267)
	at org.apache.uima.analysis_engine.impl.PrimitiveAnalysisEngine_impl.initialize(PrimitiveAnalysisEngine_impl.java:172)
	at org.apache.uima.impl.AnalysisEngineFactory_impl.produceResource(AnalysisEngineFactory_impl.java:94)
	at org.apache.uima.impl.CompositeResourceFactory_impl.produceResource(CompositeResourceFactory_impl.java:62)
	at org.apache.uima.UIMAFramework.produceResource(UIMAFramework.java:279)
	at org.apache.uima.UIMAFramework.produceAnalysisEngine(UIMAFramework.java:407)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl.setup(ASB_impl.java:256)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.initASB(AggregateAnalysisEngine_impl.java:435)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.initializeAggregateAnalysisEngine(AggregateAnalysisEngine_impl.java:379)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.initialize(AggregateAnalysisEngine_impl.java:192)
	at org.apache.uima.impl.AnalysisEngineFactory_impl.produceResource(AnalysisEngineFactory_impl.java:94)
	at org.apache.uima.impl.CompositeResourceFactory_impl.produceResource(CompositeResourceFactory_impl.java:62)
	at org.apache.uima.UIMAFramework.produceResource(UIMAFramework.java:279)
	at org.apache.uima.UIMAFramework.produceAnalysisEngine(UIMAFramework.java:407)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl.setup(ASB_impl.java:256)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.initASB(AggregateAnalysisEngine_impl.java:435)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.initializeAggregateAnalysisEngine(AggregateAnalysisEngine_impl.java:379)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.initialize(AggregateAnalysisEngine_impl.java:192)
	at org.apache.uima.impl.AnalysisEngineFactory_impl.produceResource(AnalysisEngineFactory_impl.java:94)
	at org.apache.uima.impl.CompositeResourceFactory_impl.produceResource(CompositeResourceFactory_impl.java:62)
	at org.apache.uima.UIMAFramework.produceResource(UIMAFramework.java:279)
	at org.apache.uima.UIMAFramework.produceResource(UIMAFramework.java:331)
	at org.apache.uima.UIMAFramework.produceAnalysisEngine(UIMAFramework.java:448)
	at org.ohnlp.util.SimpleCliPipeline.runPipeline(SimpleCliPipeline.java:50)
	at org.ohnlp.medxn.Main.main(Main.java:39)
org.apache.uima.resource.ResourceAccessException
	at org.apache.uima.impl.UimaContext_ImplBase.getResourceFilePath(UimaContext_ImplBase.java:359)
	at org.ohnlp.medxn.ae.MedNormRxCUIAnnotator.initialize(MedNormRxCUIAnnotator.java:56)
	at org.apache.uima.analysis_engine.impl.PrimitiveAnalysisEngine_impl.initializeAnalysisComponent(PrimitiveAnalysisEngine_impl.java:267)
	at org.apache.uima.analysis_engine.impl.PrimitiveAnalysisEngine_impl.initialize(PrimitiveAnalysisEngine_impl.java:172)
	at org.apache.uima.impl.AnalysisEngineFactory_impl.produceResource(AnalysisEngineFactory_impl.java:94)
	at org.apache.uima.impl.CompositeResourceFactory_impl.produceResource(CompositeResourceFactory_impl.java:62)
	at org.apache.uima.UIMAFramework.produceResource(UIMAFramework.java:279)
	at org.apache.uima.UIMAFramework.produceAnalysisEngine(UIMAFramework.java:407)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl.setup(ASB_impl.java:256)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.initASB(AggregateAnalysisEngine_impl.java:435)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.initializeAggregateAnalysisEngine(AggregateAnalysisEngine_impl.java:379)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.initialize(AggregateAnalysisEngine_impl.java:192)
	at org.apache.uima.impl.AnalysisEngineFactory_impl.produceResource(AnalysisEngineFactory_impl.java:94)
	at org.apache.uima.impl.CompositeResourceFactory_impl.produceResource(CompositeResourceFactory_impl.java:62)
	at org.apache.uima.UIMAFramework.produceResource(UIMAFramework.java:279)
	at org.apache.uima.UIMAFramework.produceAnalysisEngine(UIMAFramework.java:407)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl.setup(ASB_impl.java:256)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.initASB(AggregateAnalysisEngine_impl.java:435)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.initializeAggregateAnalysisEngine(AggregateAnalysisEngine_impl.java:379)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.initialize(AggregateAnalysisEngine_impl.java:192)
	at org.apache.uima.impl.AnalysisEngineFactory_impl.produceResource(AnalysisEngineFactory_impl.java:94)
	at org.apache.uima.impl.CompositeResourceFactory_impl.produceResource(CompositeResourceFactory_impl.java:62)
	at org.apache.uima.UIMAFramework.produceResource(UIMAFramework.java:279)
	at org.apache.uima.UIMAFramework.produceResource(UIMAFramework.java:331)
	at org.apache.uima.UIMAFramework.produceAnalysisEngine(UIMAFramework.java:448)
	at org.ohnlp.util.SimpleCliPipeline.runPipeline(SimpleCliPipeline.java:50)
	at org.ohnlp.medxn.Main.main(Main.java:39)
org.apache.uima.resource.ResourceAccessException
	at org.apache.uima.impl.UimaContext_ImplBase.getResourceFilePath(UimaContext_ImplBase.java:359)
	at org.ohnlp.medxn.ae.ACLookupRxCUIDrugNormAnnotator.initialize(ACLookupRxCUIDrugNormAnnotator.java:66)
	at org.apache.uima.analysis_engine.impl.PrimitiveAnalysisEngine_impl.initializeAnalysisComponent(PrimitiveAnalysisEngine_impl.java:267)
	at org.apache.uima.analysis_engine.impl.PrimitiveAnalysisEngine_impl.initialize(PrimitiveAnalysisEngine_impl.java:172)
	at org.apache.uima.impl.AnalysisEngineFactory_impl.produceResource(AnalysisEngineFactory_impl.java:94)
	at org.apache.uima.impl.CompositeResourceFactory_impl.produceResource(CompositeResourceFactory_impl.java:62)
	at org.apache.uima.UIMAFramework.produceResource(UIMAFramework.java:279)
	at org.apache.uima.UIMAFramework.produceAnalysisEngine(UIMAFramework.java:407)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl.setup(ASB_impl.java:256)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.initASB(AggregateAnalysisEngine_impl.java:435)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.initializeAggregateAnalysisEngine(AggregateAnalysisEngine_impl.java:379)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.initialize(AggregateAnalysisEngine_impl.java:192)
	at org.apache.uima.impl.AnalysisEngineFactory_impl.produceResource(AnalysisEngineFactory_impl.java:94)
	at org.apache.uima.impl.CompositeResourceFactory_impl.produceResource(CompositeResourceFactory_impl.java:62)
	at org.apache.uima.UIMAFramework.produceResource(UIMAFramework.java:279)
	at org.apache.uima.UIMAFramework.produceAnalysisEngine(UIMAFramework.java:407)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl.setup(ASB_impl.java:256)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.initASB(AggregateAnalysisEngine_impl.java:435)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.initializeAggregateAnalysisEngine(AggregateAnalysisEngine_impl.java:379)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.initialize(AggregateAnalysisEngine_impl.java:192)
	at org.apache.uima.impl.AnalysisEngineFactory_impl.produceResource(AnalysisEngineFactory_impl.java:94)
	at org.apache.uima.impl.CompositeResourceFactory_impl.produceResource(CompositeResourceFactory_impl.java:62)
	at org.apache.uima.UIMAFramework.produceResource(UIMAFramework.java:279)
	at org.apache.uima.UIMAFramework.produceResource(UIMAFramework.java:331)
	at org.apache.uima.UIMAFramework.produceAnalysisEngine(UIMAFramework.java:448)
	at org.ohnlp.util.SimpleCliPipeline.runPipeline(SimpleCliPipeline.java:50)
	at org.ohnlp.medxn.Main.main(Main.java:39)
2021-02-09 21:20:02 INFO  LvgLookup:144 - loading LVG condensed dictionary from dictStream
2021-02-09 21:20:02 INFO  LvgLookup:162 - loaded resource, lines=87272
org.apache.uima.resource.ResourceAccessException
	at org.apache.uima.impl.UimaContext_ImplBase.getResourceFilePath(UimaContext_ImplBase.java:359)
	at org.ohnlp.medxn.ae.ACLookupDrugAnnotator.initialize(ACLookupDrugAnnotator.java:70)
	at org.apache.uima.analysis_engine.impl.PrimitiveAnalysisEngine_impl.initializeAnalysisComponent(PrimitiveAnalysisEngine_impl.java:267)
	at org.apache.uima.analysis_engine.impl.PrimitiveAnalysisEngine_impl.initialize(PrimitiveAnalysisEngine_impl.java:172)
	at org.apache.uima.impl.AnalysisEngineFactory_impl.produceResource(AnalysisEngineFactory_impl.java:94)
	at org.apache.uima.impl.CompositeResourceFactory_impl.produceResource(CompositeResourceFactory_impl.java:62)
	at org.apache.uima.UIMAFramework.produceResource(UIMAFramework.java:279)
	at org.apache.uima.UIMAFramework.produceAnalysisEngine(UIMAFramework.java:407)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl.setup(ASB_impl.java:256)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.initASB(AggregateAnalysisEngine_impl.java:435)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.initializeAggregateAnalysisEngine(AggregateAnalysisEngine_impl.java:379)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.initialize(AggregateAnalysisEngine_impl.java:192)
	at org.apache.uima.impl.AnalysisEngineFactory_impl.produceResource(AnalysisEngineFactory_impl.java:94)
	at org.apache.uima.impl.CompositeResourceFactory_impl.produceResource(CompositeResourceFactory_impl.java:62)
	at org.apache.uima.UIMAFramework.produceResource(UIMAFramework.java:279)
	at org.apache.uima.UIMAFramework.produceAnalysisEngine(UIMAFramework.java:407)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl.setup(ASB_impl.java:256)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.initASB(AggregateAnalysisEngine_impl.java:435)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.initializeAggregateAnalysisEngine(AggregateAnalysisEngine_impl.java:379)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.initialize(AggregateAnalysisEngine_impl.java:192)
	at org.apache.uima.impl.AnalysisEngineFactory_impl.produceResource(AnalysisEngineFactory_impl.java:94)
	at org.apache.uima.impl.CompositeResourceFactory_impl.produceResource(CompositeResourceFactory_impl.java:62)
	at org.apache.uima.UIMAFramework.produceResource(UIMAFramework.java:279)
	at org.apache.uima.UIMAFramework.produceResource(UIMAFramework.java:331)
	at org.apache.uima.UIMAFramework.produceAnalysisEngine(UIMAFramework.java:448)
	at org.ohnlp.util.SimpleCliPipeline.runPipeline(SimpleCliPipeline.java:50)
	at org.ohnlp.medxn.Main.main(Main.java:39)
Feb 09, 2021 9:20:02 PM opennlp.uima.postag.POSTagger initialize(111)
INFO: Initializing the OpenNLP Part of Speech annotator.
Feb 09, 2021 9:20:02 PM opennlp.uima.util.AnnotatorUtil getOptionalParameter(422)
INFO: opennlp.uima.BeamSize = not set
Feb 09, 2021 9:20:02 PM opennlp.uima.tokenize.AbstractTokenizer initialize(71)
INFO: Initializing the OpenNLP Tokenizer annotator.
Feb 09, 2021 9:20:02 PM opennlp.uima.util.AnnotatorUtil getOptionalParameter(422)
INFO: opennlp.uima.IsRemoveExistingAnnotations = not set
Feb 09, 2021 9:20:02 PM opennlp.uima.sentdetect.AbstractSentenceDetector initialize(61)
INFO: Initializing the OpenNLP Sentence annotator.
Feb 09, 2021 9:20:02 PM opennlp.uima.util.AnnotatorUtil getOptionalParameter(422)
INFO: opennlp.uima.IsRemoveExistingAnnotations = not set
Feb 09, 2021 9:20:03 PM opennlp.uima.util.AnnotatorUtil getOptionalParameter(422)
INFO: opennlp.uima.ContainerType = not set
Feb 09, 2021 9:20:03 PM opennlp.uima.util.AnnotatorUtil getOptionalParameter(422)
INFO: opennlp.uima.SentenceType = org.ohnlp.typesystem.type.textspan.Sentence
Feb 09, 2021 9:20:03 PM opennlp.uima.util.AnnotatorUtil getOptionalParameter(422)
INFO: opennlp.uima.ProbabilityFeature = not set
Feb 09, 2021 9:20:03 PM opennlp.uima.util.AnnotatorUtil getOptionalParameter(422)
INFO: opennlp.uima.SentenceType = org.ohnlp.typesystem.type.textspan.Sentence
Feb 09, 2021 9:20:03 PM opennlp.uima.util.AnnotatorUtil getOptionalParameter(422)
INFO: opennlp.uima.TokenType = org.ohnlp.typesystem.type.syntax.BaseToken
Feb 09, 2021 9:20:03 PM opennlp.uima.util.AnnotatorUtil getOptionalParameter(422)
INFO: opennlp.uima.ProbabilityFeature = not set
Feb 09, 2021 9:20:03 PM opennlp.uima.util.AnnotatorUtil getOptionalParameter(422)
INFO: opennlp.uima.SentenceType = org.ohnlp.typesystem.type.textspan.Sentence
Feb 09, 2021 9:20:03 PM opennlp.uima.util.AnnotatorUtil getOptionalParameter(422)
INFO: opennlp.uima.TokenType = org.ohnlp.typesystem.type.syntax.BaseToken
Feb 09, 2021 9:20:03 PM opennlp.uima.util.AnnotatorUtil getOptionalParameter(422)
INFO: opennlp.uima.POSFeature = partOfSpeech
Feb 09, 2021 9:20:03 PM opennlp.uima.util.AnnotatorUtil getOptionalParameter(422)
INFO: opennlp.uima.ProbabilityFeature = not set
Feb 09, 2021 9:20:03 PM opennlp.uima.util.AnnotatorUtil getOptionalParameter(422)
INFO: opennlp.uima.ChunkType = org.ohnlp.typesystem.type.syntax.Chunk
Feb 09, 2021 9:20:03 PM opennlp.uima.util.AnnotatorUtil getOptionalParameter(422)
INFO: opennlp.uima.ChunkTagFeature = chunkType
Feb 09, 2021 9:20:03 PM opennlp.uima.util.AnnotatorUtil getOptionalParameter(422)
INFO: opennlp.uima.TokenType = org.ohnlp.typesystem.type.syntax.BaseToken
Feb 09, 2021 9:20:03 PM opennlp.uima.util.AnnotatorUtil getOptionalParameter(422)
INFO: opennlp.uima.POSFeature = partOfSpeech
Feb 09, 2021 9:20:03 PM org.apache.uima.analysis_engine.impl.PrimitiveAnalysisEngine_impl callAnalysisComponentProcess(434)
SEVERE: Exception occurred
org.apache.uima.analysis_engine.AnalysisEngineProcessException: Annotator processing failed.    
	at org.apache.uima.analysis_engine.impl.PrimitiveAnalysisEngine_impl.callAnalysisComponentProcess(PrimitiveAnalysisEngine_impl.java:416)
	at org.apache.uima.analysis_engine.impl.PrimitiveAnalysisEngine_impl.processAndOutputNewCASes(PrimitiveAnalysisEngine_impl.java:318)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl$AggregateCasIterator.processUntilNextOutputCas(ASB_impl.java:570)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl$AggregateCasIterator.<init>(ASB_impl.java:412)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl.process(ASB_impl.java:344)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.processAndOutputNewCASes(AggregateAnalysisEngine_impl.java:271)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl$AggregateCasIterator.processUntilNextOutputCas(ASB_impl.java:570)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl$AggregateCasIterator.<init>(ASB_impl.java:412)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl.process(ASB_impl.java:344)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.processAndOutputNewCASes(AggregateAnalysisEngine_impl.java:271)
	at org.apache.uima.analysis_engine.impl.AnalysisEngineImplBase.process(AnalysisEngineImplBase.java:269)
	at org.ohnlp.util.SimpleCliPipeline.runPipeline(SimpleCliPipeline.java:57)
	at org.ohnlp.medxn.Main.main(Main.java:39)
Caused by: java.lang.NullPointerException
	at org.ohnlp.medxn.ae.ACLookupDrugAnnotator.process(ACLookupDrugAnnotator.java:129)
	at org.apache.uima.analysis_component.JCasAnnotator_ImplBase.process(JCasAnnotator_ImplBase.java:48)
	at org.apache.uima.analysis_engine.impl.PrimitiveAnalysisEngine_impl.callAnalysisComponentProcess(PrimitiveAnalysisEngine_impl.java:401)
	... 12 more

Feb 09, 2021 9:20:03 PM org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl processAndOutputNewCASes(279)
SEVERE: Exception occurred
org.apache.uima.analysis_engine.AnalysisEngineProcessException: Annotator processing failed.    
	at org.apache.uima.analysis_engine.impl.PrimitiveAnalysisEngine_impl.callAnalysisComponentProcess(PrimitiveAnalysisEngine_impl.java:416)
	at org.apache.uima.analysis_engine.impl.PrimitiveAnalysisEngine_impl.processAndOutputNewCASes(PrimitiveAnalysisEngine_impl.java:318)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl$AggregateCasIterator.processUntilNextOutputCas(ASB_impl.java:570)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl$AggregateCasIterator.<init>(ASB_impl.java:412)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl.process(ASB_impl.java:344)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.processAndOutputNewCASes(AggregateAnalysisEngine_impl.java:271)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl$AggregateCasIterator.processUntilNextOutputCas(ASB_impl.java:570)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl$AggregateCasIterator.<init>(ASB_impl.java:412)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl.process(ASB_impl.java:344)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.processAndOutputNewCASes(AggregateAnalysisEngine_impl.java:271)
	at org.apache.uima.analysis_engine.impl.AnalysisEngineImplBase.process(AnalysisEngineImplBase.java:269)
	at org.ohnlp.util.SimpleCliPipeline.runPipeline(SimpleCliPipeline.java:57)
	at org.ohnlp.medxn.Main.main(Main.java:39)
Caused by: java.lang.NullPointerException
	at org.ohnlp.medxn.ae.ACLookupDrugAnnotator.process(ACLookupDrugAnnotator.java:129)
	at org.apache.uima.analysis_component.JCasAnnotator_ImplBase.process(JCasAnnotator_ImplBase.java:48)
	at org.apache.uima.analysis_engine.impl.PrimitiveAnalysisEngine_impl.callAnalysisComponentProcess(PrimitiveAnalysisEngine_impl.java:401)
	... 12 more

Feb 09, 2021 9:20:03 PM org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl processAndOutputNewCASes(279)
SEVERE: Exception occurred
org.apache.uima.analysis_engine.AnalysisEngineProcessException: Annotator processing failed.    
	at org.apache.uima.analysis_engine.impl.PrimitiveAnalysisEngine_impl.callAnalysisComponentProcess(PrimitiveAnalysisEngine_impl.java:416)
	at org.apache.uima.analysis_engine.impl.PrimitiveAnalysisEngine_impl.processAndOutputNewCASes(PrimitiveAnalysisEngine_impl.java:318)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl$AggregateCasIterator.processUntilNextOutputCas(ASB_impl.java:570)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl$AggregateCasIterator.<init>(ASB_impl.java:412)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl.process(ASB_impl.java:344)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.processAndOutputNewCASes(AggregateAnalysisEngine_impl.java:271)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl$AggregateCasIterator.processUntilNextOutputCas(ASB_impl.java:570)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl$AggregateCasIterator.<init>(ASB_impl.java:412)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl.process(ASB_impl.java:344)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.processAndOutputNewCASes(AggregateAnalysisEngine_impl.java:271)
	at org.apache.uima.analysis_engine.impl.AnalysisEngineImplBase.process(AnalysisEngineImplBase.java:269)
	at org.ohnlp.util.SimpleCliPipeline.runPipeline(SimpleCliPipeline.java:57)
	at org.ohnlp.medxn.Main.main(Main.java:39)
Caused by: java.lang.NullPointerException
	at org.ohnlp.medxn.ae.ACLookupDrugAnnotator.process(ACLookupDrugAnnotator.java:129)
	at org.apache.uima.analysis_component.JCasAnnotator_ImplBase.process(JCasAnnotator_ImplBase.java:48)
	at org.apache.uima.analysis_engine.impl.PrimitiveAnalysisEngine_impl.callAnalysisComponentProcess(PrimitiveAnalysisEngine_impl.java:401)
	... 12 more

Exception in thread "main" org.apache.uima.analysis_engine.AnalysisEngineProcessException: Annotator processing failed.    
	at org.apache.uima.analysis_engine.impl.PrimitiveAnalysisEngine_impl.callAnalysisComponentProcess(PrimitiveAnalysisEngine_impl.java:416)
	at org.apache.uima.analysis_engine.impl.PrimitiveAnalysisEngine_impl.processAndOutputNewCASes(PrimitiveAnalysisEngine_impl.java:318)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl$AggregateCasIterator.processUntilNextOutputCas(ASB_impl.java:570)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl$AggregateCasIterator.<init>(ASB_impl.java:412)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl.process(ASB_impl.java:344)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.processAndOutputNewCASes(AggregateAnalysisEngine_impl.java:271)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl$AggregateCasIterator.processUntilNextOutputCas(ASB_impl.java:570)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl$AggregateCasIterator.<init>(ASB_impl.java:412)
	at org.apache.uima.analysis_engine.asb.impl.ASB_impl.process(ASB_impl.java:344)
	at org.apache.uima.analysis_engine.impl.AggregateAnalysisEngine_impl.processAndOutputNewCASes(AggregateAnalysisEngine_impl.java:271)
	at org.apache.uima.analysis_engine.impl.AnalysisEngineImplBase.process(AnalysisEngineImplBase.java:269)
	at org.ohnlp.util.SimpleCliPipeline.runPipeline(SimpleCliPipeline.java:57)
	at org.ohnlp.medxn.Main.main(Main.java:39)
Caused by: java.lang.NullPointerException
	at org.ohnlp.medxn.ae.ACLookupDrugAnnotator.process(ACLookupDrugAnnotator.java:129)
	at org.apache.uima.analysis_component.JCasAnnotator_ImplBase.process(JCasAnnotator_ImplBase.java:48)
	at org.apache.uima.analysis_engine.impl.PrimitiveAnalysisEngine_impl.callAnalysisComponentProcess(PrimitiveAnalysisEngine_impl.java:401)
	... 12 more

--- a/MedXN.iml
+++ b/MedXN.iml
@@ -1,5 +1,10 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <module org.jetbrains.idea.maven.project.MavenProjectsManager.isMavenModule="true" type="JAVA_MODULE" version="4">
+  <component name="FacetManager">
+    <facet type="Spring" name="Spring">
+      <configuration />
+    </facet>
+  </component>
   <component name="NewModuleRootManager" LANGUAGE_LEVEL="JDK_1_8">
     <output url="file://$MODULE_DIR$/target/classes" />
     <output-test url="file://$MODULE_DIR$/target/test-classes" />
@@ -10,6 +15,61 @@
     </content>
     <orderEntry type="inheritedJdk" />
     <orderEntry type="sourceFolder" forTests="false" />
-    <orderEntry type="library" name="Maven: org.ohnlp.medtagger:MedTagger:1.0.3" level="project" />
+    <orderEntry type="library" name="Maven: org.ohnlp.medtagger:MedTagger:1.0.9" level="project" />
+    <orderEntry type="library" name="Maven: commons-io:commons-io:2.5" level="project" />
+    <orderEntry type="library" name="Maven: org.apache.commons:commons-lang3:3.9" level="project" />
+    <orderEntry type="library" name="Maven: commons-lang:commons-lang:2.6" level="project" />
+    <orderEntry type="library" name="Maven: commons-logging:commons-logging:1.2" level="project" />
+    <orderEntry type="library" name="Maven: net.sf.jwordnet:jwnl:1.3.3" level="project" />
+    <orderEntry type="library" name="Maven: log4j:log4j:1.2.17" level="project" />
+    <orderEntry type="library" name="Maven: org.apache.opennlp:opennlp-maxent:3.0.3" level="project" />
+    <orderEntry type="library" name="Maven: org.apache.opennlp:opennlp-tools:1.9.0" level="project" />
+    <orderEntry type="library" name="Maven: org.apache.opennlp:opennlp-uima:1.9.0" level="project" />
+    <orderEntry type="library" name="Maven: org.apache.uima:uimaj-adapter-vinci:2.10.0" level="project" />
+    <orderEntry type="library" name="Maven: org.apache.uima:jVinci:2.10.0" level="project" />
+    <orderEntry type="library" name="Maven: org.apache.uima:uimaj-core:2.10.0" level="project" />
+    <orderEntry type="library" name="Maven: org.apache.uima:uimafit-core:2.4.0" level="project" />
+    <orderEntry type="library" name="Maven: commons-logging:commons-logging-api:1.1" level="project" />
+    <orderEntry type="library" name="Maven: org.springframework:spring-core:3.2.16.RELEASE" level="project" />
+    <orderEntry type="library" name="Maven: org.springframework:spring-context:3.2.16.RELEASE" level="project" />
+    <orderEntry type="library" name="Maven: org.springframework:spring-aop:3.2.16.RELEASE" level="project" />
+    <orderEntry type="library" name="Maven: aopalliance:aopalliance:1.0" level="project" />
+    <orderEntry type="library" name="Maven: org.springframework:spring-expression:3.2.16.RELEASE" level="project" />
+    <orderEntry type="library" name="Maven: org.springframework:spring-beans:3.2.16.RELEASE" level="project" />
+    <orderEntry type="library" name="Maven: org.apache.uima:uimaj-cpe:2.10.0" level="project" />
+    <orderEntry type="library" name="Maven: org.apache.uima:uimaj-document-annotation:2.10.0" level="project" />
+    <orderEntry type="library" name="Maven: org.apache.uima:uimaj-tools:2.10.0" level="project" />
+    <orderEntry type="library" name="Maven: org.ahocorasick:ahocorasick:0.4.0" level="project" />
+    <orderEntry type="library" name="Maven: me.tongfei:progressbar:0.7.4" level="project" />
+    <orderEntry type="library" name="Maven: org.jline:jline:3.11.0" level="project" />
+    <orderEntry type="library" name="Maven: org.fusesource.jansi:jansi:1.18" level="project" />
+    <orderEntry type="library" scope="PROVIDED" name="Maven: org.ohnlp.backbone:API:1.0.0-SNAPSHOT" level="project" />
+    <orderEntry type="library" scope="PROVIDED" name="Maven: org.apache.beam:beam-sdks-java-core:2.22.0" level="project" />
+    <orderEntry type="library" scope="PROVIDED" name="Maven: org.apache.beam:beam-model-pipeline:2.22.0" level="project" />
+    <orderEntry type="library" scope="PROVIDED" name="Maven: org.apache.beam:beam-vendor-grpc-1_26_0:0.3" level="project" />
+    <orderEntry type="library" scope="PROVIDED" name="Maven: com.google.errorprone:error_prone_annotations:2.3.3" level="project" />
+    <orderEntry type="library" scope="PROVIDED" name="Maven: org.apache.logging.log4j:log4j-api:2.6.2" level="project" />
+    <orderEntry type="library" scope="PROVIDED" name="Maven: org.conscrypt:conscrypt-openjdk-uber:1.3.0" level="project" />
+    <orderEntry type="library" scope="PROVIDED" name="Maven: org.apache.beam:beam-model-job-management:2.22.0" level="project" />
+    <orderEntry type="library" scope="PROVIDED" name="Maven: org.apache.beam:beam-vendor-bytebuddy-1_10_8:0.1" level="project" />
+    <orderEntry type="library" scope="PROVIDED" name="Maven: org.apache.beam:beam-vendor-guava-26_0-jre:0.1" level="project" />
+    <orderEntry type="library" scope="PROVIDED" name="Maven: com.google.code.findbugs:jsr305:3.0.2" level="project" />
+    <orderEntry type="library" scope="PROVIDED" name="Maven: com.fasterxml.jackson.core:jackson-core:2.10.2" level="project" />
+    <orderEntry type="library" scope="PROVIDED" name="Maven: com.fasterxml.jackson.core:jackson-annotations:2.10.2" level="project" />
+    <orderEntry type="library" scope="PROVIDED" name="Maven: com.fasterxml.jackson.core:jackson-databind:2.10.2" level="project" />
+    <orderEntry type="library" scope="PROVIDED" name="Maven: org.slf4j:slf4j-api:1.7.25" level="project" />
+    <orderEntry type="library" scope="PROVIDED" name="Maven: org.apache.avro:avro:1.8.2" level="project" />
+    <orderEntry type="library" scope="PROVIDED" name="Maven: org.codehaus.jackson:jackson-core-asl:1.9.13" level="project" />
+    <orderEntry type="library" scope="PROVIDED" name="Maven: org.codehaus.jackson:jackson-mapper-asl:1.9.13" level="project" />
+    <orderEntry type="library" scope="PROVIDED" name="Maven: com.thoughtworks.paranamer:paranamer:2.7" level="project" />
+    <orderEntry type="library" scope="PROVIDED" name="Maven: org.apache.commons:commons-compress:1.8.1" level="project" />
+    <orderEntry type="library" scope="PROVIDED" name="Maven: org.xerial.snappy:snappy-java:1.1.4" level="project" />
+    <orderEntry type="library" scope="PROVIDED" name="Maven: joda-time:joda-time:2.10.5" level="project" />
+    <orderEntry type="library" scope="PROVIDED" name="Maven: org.tukaani:xz:1.8" level="project" />
+    <orderEntry type="library" scope="PROVIDED" name="Maven: org.apache.beam:beam-runners-core-java:2.22.0" level="project" />
+    <orderEntry type="library" scope="PROVIDED" name="Maven: org.apache.beam:beam-model-fn-execution:2.22.0" level="project" />
+    <orderEntry type="library" scope="PROVIDED" name="Maven: org.apache.beam:beam-runners-core-construction-java:2.22.0" level="project" />
+    <orderEntry type="library" scope="PROVIDED" name="Maven: io.github.classgraph:classgraph:4.8.65" level="project" />
+    <orderEntry type="library" scope="PROVIDED" name="Maven: org.apache.beam:beam-sdks-java-fn-execution:2.22.0" level="project" />
   </component>
 </module>

--- a/README.md
+++ b/README.md
@@ -16,23 +16,36 @@ If you do not have the prerequisites installed, you may need to download the bin
 download sites and add the unzipped bin directories to `PATH`. These steps are dependent on your OS.
 
 ## Installation 
+MedXN is runnable out-of-the-box: simply download MedXN.zip corresponding to the
+ [latest release](https://github.com/OHNLP/MedXN/releases/), extract, and follow the instructions for running 
 
-To install MedXN, first clone this repo and enter the project directory as root:
+## Compiling From Source Code
+To make changes and re-compile medxn from source code, first clone this repo and enter the project directory as root:
 ```
 git clone https://github.com/OHNLP/MedXN.git
 cd MedXN
 git checkout dist
 ```
 
-The build.xml contains an Apache Ant script to download MedTagger 1.0.3 automatically, assume you have Ant installed, run:
+To download MedTagger and Backbone dependencies from github packages, you will have to add the following to your
+maven settings.xml (typically located in ~/.m2/settings.xml)
+
+```xml
+ <servers>
+        <server>
+            <id>medtagger</id>
+            <username>your_github_username</username>
+            <password>your_github_access_token</password>
+        </server>
+        <server>
+            <id>backbone-maven</id>
+            <username>your_github_username</username>
+            <password>your_github_access_token</password>
+        </server>
+    </servers>
 ```
-ant download_medtagger
-```
-This will download a `MedTagger.jar` to a newly created `dist` directory. 
-Then, install MedTagger v1.0.3 in your local maven repo (typically `.m2`):
-```
-mvn install:install-file -Dfile=dist/MedTagger.jar -DgroupId=org.ohnlp.medtagger -DartifactId=MedTagger -Dversion=1.0.3 -Dpackaging=jar
-```
+where your github access token corresponds to an appropriate 
+[github token](https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line) with the read:packages permissions
 
 To build MedXN, build the jar file using:
 ```

--- a/README.md
+++ b/README.md
@@ -185,3 +185,5 @@ Under `dist/resources/medxnresources`
 
 `regExPatterns.txt`: contains medication attribute patterns written in Java regular expression (includes usage descriptions).
 
+
+

--- a/pom.xml
+++ b/pom.xml
@@ -5,8 +5,8 @@
     <modelVersion>4.0.0</modelVersion>
 
     <groupId>org.ohnlp.medxn</groupId>
-    <artifactId>MedXN</artifactId>
-    <version>1.0.3</version>
+    <artifactId>medxn</artifactId>
+    <version>1.0.4</version>
     <description>The MedXN medication Information extraction pipeline</description>
 
     <repositories>
@@ -14,6 +14,11 @@
             <id>medtagger</id>
             <name>GitHub OHNLP MedTagger Apache Maven Packages</name>
             <url>https://maven.pkg.github.com/OHNLP/MedTagger</url>
+        </repository>
+        <repository>
+            <id>backbone-maven</id>
+            <name>OHNLP Backbone Apache Maven Packages</name>
+            <url>https://maven.pkg.github.com/OHNLP/Backbone</url>
         </repository>
     </repositories>
 
@@ -74,7 +79,13 @@
         <dependency>
             <groupId>org.ohnlp.medtagger</groupId>
             <artifactId>MedTagger</artifactId>
-            <version>1.0.7</version>
+            <version>1.0.9</version>
+        </dependency>
+        <dependency>
+            <groupId>org.ohnlp.backbone</groupId>
+            <artifactId>API</artifactId>
+            <version>1.0.0-SNAPSHOT</version>
+            <scope>provided</scope>
         </dependency>
     </dependencies>
 

--- a/settings.xml
+++ b/settings.xml
@@ -8,5 +8,10 @@
             <username>${env.SECRET_ACTOR}</username>
             <password>${env.SECRET_TOKEN}</password>
         </server>
+        <server>
+            <id>backbone-maven</id>
+            <username>${env.SECRET_ACTOR}</username>
+            <password>${env.SECRET_TOKEN}</password>
+        </server>
     </servers>
 </settings>

--- a/src/main/java/org/ohnlp/medxn/ae/ACLookupDrugAnnotator.java
+++ b/src/main/java/org/ohnlp/medxn/ae/ACLookupDrugAnnotator.java
@@ -66,12 +66,10 @@ public class ACLookupDrugAnnotator extends JCasAnnotator_ImplBase {
 
 
 		try {
-
-			String dict = aContext.getResourceFilePath("RxNorm_BNIN");
-			btac = new AhoCorasickDict(dict);	
+			btac = new AhoCorasickDict(aContext.getResourceAsStream("RxNorm_BNIN"));
 		} catch (ResourceAccessException e) {
 			e.printStackTrace();
-		} 		
+		}
 	}
 
 	@Override

--- a/src/main/java/org/ohnlp/medxn/ae/ACLookupDrugNormAnnotator.java
+++ b/src/main/java/org/ohnlp/medxn/ae/ACLookupDrugNormAnnotator.java
@@ -59,14 +59,12 @@ public class ACLookupDrugNormAnnotator extends JCasAnnotator_ImplBase {
 			throws ResourceInitializationException {
 		super.initialize(aContext);
 		logger.setLevel(Level.DEBUG);
-				
+
 		try {
-			
-			String dict = aContext.getResourceFilePath("RxNorm_Name");
-			btac = new AhoCorasickDict(dict);		
+			btac = new AhoCorasickDict(aContext.getResourceAsStream("RxNorm_Name"));
 		} catch (ResourceAccessException e) {
 			e.printStackTrace();
-		} 		
+		}
 	}
 
 	@Override

--- a/src/main/java/org/ohnlp/medxn/ae/ACLookupRxCUIDrugNormAnnotator.java
+++ b/src/main/java/org/ohnlp/medxn/ae/ACLookupRxCUIDrugNormAnnotator.java
@@ -62,9 +62,7 @@ public class ACLookupRxCUIDrugNormAnnotator extends JCasAnnotator_ImplBase {
 		logger.setLevel(Level.DEBUG);
 				
 		try {
-			
-			String dict = aContext.getResourceFilePath("RxCUI");
-			btac = new AhoCorasickDict(dict);		
+			btac = new AhoCorasickDict( aContext.getResourceAsStream("RxCUI"));
 		} catch (ResourceAccessException e) {
 			e.printStackTrace();
 		} 		

--- a/src/main/java/org/ohnlp/medxn/ae/MedNormRxCUIAnnotator.java
+++ b/src/main/java/org/ohnlp/medxn/ae/MedNormRxCUIAnnotator.java
@@ -53,10 +53,9 @@ public class MedNormRxCUIAnnotator extends JCasAnnotator_ImplBase {
 		doseRxCUI = new HashMap<String, String>();
 		
 		try {
-			String dictfile = uimaContext.getResourceFilePath("doseDict");
 			BufferedReader fileReader = null;
 			try {
-				fileReader = new BufferedReader(new InputStreamReader(new FileInputStream(dictfile)));
+				fileReader = new BufferedReader(new InputStreamReader(uimaContext.getResourceAsStream("doseDict")));
 
 				String line="";
 				while((line=fileReader.readLine()) != null) {

--- a/src/main/java/org/ohnlp/medxn/backbone/MedXNBackboneTransform.java
+++ b/src/main/java/org/ohnlp/medxn/backbone/MedXNBackboneTransform.java
@@ -70,7 +70,7 @@ public class MedXNBackboneTransform extends Transform {
         @Setup
         public void init() throws IOException, InvalidXMLException, ResourceInitializationException {
             AnalysisEngineDescription aaeDesc = createEngineDescription(
-                    "desc.medtaggeriedesc.aggregate_analysis_engine.MedTaggerIEAggregateTAE");
+                    "desc.medxndesc.aggregate_analysis_engine.MedXNAggregateTAE");
             this.resMgr = ResourceManagerFactory.newResourceManager();
             this.aae = UIMAFramework.produceAnalysisEngine(aaeDesc, resMgr, null);
             this.cas = CasCreationUtils.createCas(Collections.singletonList(aae.getMetaData()),

--- a/src/main/java/org/ohnlp/medxn/backbone/MedXNBackboneTransform.java
+++ b/src/main/java/org/ohnlp/medxn/backbone/MedXNBackboneTransform.java
@@ -10,7 +10,6 @@ import org.apache.uima.UIMAFramework;
 import org.apache.uima.analysis_engine.AnalysisEngine;
 import org.apache.uima.analysis_engine.AnalysisEngineDescription;
 import org.apache.uima.analysis_engine.AnalysisEngineProcessException;
-import org.apache.uima.analysis_engine.metadata.AnalysisEngineMetaData;
 import org.apache.uima.cas.CAS;
 import org.apache.uima.cas.CASException;
 import org.apache.uima.cas.FeatureStructure;
@@ -21,7 +20,6 @@ import org.apache.uima.jcas.cas.FSArray;
 import org.apache.uima.jcas.tcas.Annotation;
 import org.apache.uima.resource.ResourceInitializationException;
 import org.apache.uima.resource.ResourceManager;
-import org.apache.uima.resource.metadata.ConfigurationParameterSettings;
 import org.apache.uima.util.CasCreationUtils;
 import org.apache.uima.util.InvalidXMLException;
 import org.ohnlp.backbone.api.Transform;
@@ -32,15 +30,16 @@ import org.ohnlp.typesystem.type.textspan.Segment;
 import org.ohnlp.typesystem.type.textspan.Sentence;
 
 import java.io.IOException;
-import java.net.URI;
-import java.nio.file.FileSystem;
-import java.nio.file.FileSystemAlreadyExistsException;
-import java.nio.file.FileSystems;
+import java.text.SimpleDateFormat;
 import java.util.*;
 import java.util.stream.Collectors;
 
 import static org.apache.uima.fit.factory.AnalysisEngineFactory.createEngineDescription;
 
+/**
+ * Given an input row representing a document, duplicates row contents and adds a nlp_output_json column for each
+ * drug mention in the input text.
+ */
 public class MedXNBackboneTransform extends Transform {
     @Override
     public void initFromConfig(JsonNode jsonNode) throws ComponentInitializationException {
@@ -53,6 +52,8 @@ public class MedXNBackboneTransform extends Transform {
     }
 
     private static class MedXNPipelineFunction extends DoFn<Row, Row> {
+
+        private static ThreadLocal<SimpleDateFormat> sdf = ThreadLocal.withInitial(() -> new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssXXX"));
 
         private final String textField;
 
@@ -132,6 +133,14 @@ public class MedXNBackboneTransform extends Transform {
                     .stream()
                     .map(Annotation::getCoveredText)
                     .collect(Collectors.joining(" ")));
+            ret.setMatchSection(sectionIdx.get(drug).stream().map((s) -> {
+                try {
+                    return Integer.parseInt(s.getId());
+                } catch (Throwable var2) {
+                    return -1;
+                }
+            }).findFirst().orElse(0));
+            ret.setNlpDate(sdf.get().format(new Date()));
             return ret;
         }
 

--- a/src/main/java/org/ohnlp/medxn/backbone/MedXNBackboneTransform.java
+++ b/src/main/java/org/ohnlp/medxn/backbone/MedXNBackboneTransform.java
@@ -1,0 +1,154 @@
+package org.ohnlp.medxn.backbone;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.beam.sdk.schemas.Schema;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.Row;
+import org.apache.uima.UIMAFramework;
+import org.apache.uima.analysis_engine.AnalysisEngine;
+import org.apache.uima.analysis_engine.AnalysisEngineDescription;
+import org.apache.uima.analysis_engine.AnalysisEngineProcessException;
+import org.apache.uima.analysis_engine.metadata.AnalysisEngineMetaData;
+import org.apache.uima.cas.CAS;
+import org.apache.uima.cas.CASException;
+import org.apache.uima.cas.FeatureStructure;
+import org.apache.uima.fit.internal.ResourceManagerFactory;
+import org.apache.uima.fit.util.JCasUtil;
+import org.apache.uima.jcas.JCas;
+import org.apache.uima.jcas.cas.FSArray;
+import org.apache.uima.jcas.tcas.Annotation;
+import org.apache.uima.resource.ResourceInitializationException;
+import org.apache.uima.resource.ResourceManager;
+import org.apache.uima.resource.metadata.ConfigurationParameterSettings;
+import org.apache.uima.util.CasCreationUtils;
+import org.apache.uima.util.InvalidXMLException;
+import org.ohnlp.backbone.api.Transform;
+import org.ohnlp.backbone.api.exceptions.ComponentInitializationException;
+import org.ohnlp.medxn.type.Drug;
+import org.ohnlp.medxn.type.MedAttr;
+import org.ohnlp.typesystem.type.textspan.Segment;
+import org.ohnlp.typesystem.type.textspan.Sentence;
+
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystemAlreadyExistsException;
+import java.nio.file.FileSystems;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static org.apache.uima.fit.factory.AnalysisEngineFactory.createEngineDescription;
+
+public class MedXNBackboneTransform extends Transform {
+    @Override
+    public void initFromConfig(JsonNode jsonNode) throws ComponentInitializationException {
+        // No Configurable Initialization
+    }
+
+    @Override
+    public PCollection<Row> expand(PCollection<Row> input) {
+        return null;
+    }
+
+    private static class MedXNPipelineFunction extends DoFn<Row, Row> {
+
+        private final String textField;
+
+        // UIMA components are not serializable, and thus must be initialized per-executor via the @Setup annotation
+        private transient AnalysisEngine aae;
+        private transient ResourceManager resMgr;
+        private transient CAS cas;
+        private transient ObjectMapper om;
+
+        private MedXNPipelineFunction(String textField) {
+            this.textField = textField;
+        }
+
+        @Setup
+        public void init() throws IOException, InvalidXMLException, ResourceInitializationException {
+            AnalysisEngineDescription aaeDesc = createEngineDescription(
+                    "desc.medtaggeriedesc.aggregate_analysis_engine.MedTaggerIEAggregateTAE");
+            this.resMgr = ResourceManagerFactory.newResourceManager();
+            this.aae = UIMAFramework.produceAnalysisEngine(aaeDesc, resMgr, null);
+            this.cas = CasCreationUtils.createCas(Collections.singletonList(aae.getMetaData()),
+                    null, resMgr);
+            this.om = new ObjectMapper();
+        }
+
+        @ProcessElement
+        public void processElement(@Element Row input, OutputReceiver<Row> output) {
+            // First create the output row schema
+            List<Schema.Field> fields = new LinkedList<>(input.getSchema().getFields());
+            fields.add(Schema.Field.of("nlp_output_json", Schema.FieldType.STRING));
+            Schema schema = Schema.of(fields.toArray(new Schema.Field[0]));
+            String text = input.getString(this.textField);
+            cas.reset();
+            cas.setDocumentText(text);
+            try {
+                aae.process(cas);
+                JCas jcas = cas.getJCas();
+                Map<Drug, Collection<Sentence>> sentenceIdx = JCasUtil.indexCovering(jcas, Drug.class, Sentence.class);
+                Map<Drug, Collection<Segment>> sectionIdx = JCasUtil.indexCovering(jcas, Drug.class, Segment.class);
+                for (Drug d : JCasUtil.select(jcas, Drug.class)) {
+                    MedXNDrugBean bean = toBean(d, sentenceIdx, sectionIdx);
+                    Row out = Row.withSchema(schema).addValues(input.getValues()).addValue(om.valueToTree(bean).toString()).build();
+                    output.output(out);
+                }
+            } catch (AnalysisEngineProcessException | CASException e) {
+                e.printStackTrace();
+            }
+        }
+
+        private MedXNDrugBean toBean(Drug drug, Map<Drug, Collection<Sentence>> sentenceIdx, Map<Drug, Collection<Segment>> sectionIdx) {
+            MedXNDrugBean ret = new MedXNDrugBean();
+            // Basic Medication Attributes TODO we can do better parsing than plaintext for attrs via regex
+            String cui = drug.getNormRxCui();
+            if (cui == null) {
+                cui = drug.getName().getSemGroup();
+                if (cui != null) {
+                    cui = cui.split("::")[0];
+                }
+            }
+            ret.setCode(cui);
+            MedAttr strength = getMedAttr("strength", drug.getAttrs());
+            ret.setStrength(strength == null ? "UNSPECIFIED" : strength.getCoveredText());
+            MedAttr form = getMedAttr("form", drug.getAttrs());
+            ret.setForm(form == null ? "UNSPECIFIED" : form.getCoveredText());
+            MedAttr freq = getMedAttr("frequency", drug.getAttrs());
+            ret.setFrequency(freq == null ? "UNSPECIFIED" : freq.getCoveredText());
+            MedAttr duration = getMedAttr("duration", drug.getAttrs());
+            ret.setDuration(duration == null ? "UNSPECIFIED" : duration.getCoveredText());
+            MedAttr route = getMedAttr("route", drug.getAttrs());
+            ret.setRoute(route == null ? "UNSPECIFIED" : route.getCoveredText());
+            MedAttr dosage = getMedAttr("dosage", drug.getAttrs());
+            ret.setDosage(dosage == null ? "UNSPECIFIED" : dosage.getCoveredText());
+            // NLP Attributes
+            ret.setMatchedText(drug.getCoveredText());
+            ret.setMatchBegin(drug.getBegin());
+            ret.setMatchEnd(drug.getEnd());
+            ret.setMatchSentence( sentenceIdx.get(drug)
+                    .stream()
+                    .map(Annotation::getCoveredText)
+                    .collect(Collectors.joining(" ")));
+            return ret;
+        }
+
+        /**
+         * @param tag The name of the MedAttr to retrieve
+         * @param arr A FSArray from {@link Drug#getAttrs()}
+         * @return The attribute if it exists, or null
+         */
+        private MedAttr getMedAttr(String tag, FSArray arr) {
+            for (FeatureStructure f : arr.toArray()) {
+                if (f instanceof MedAttr) {
+                    if (((MedAttr) f).getTag().equalsIgnoreCase(tag)) {
+                        return (MedAttr) f;
+                    }
+                }
+            }
+            return null;
+        }
+    }
+}

--- a/src/main/java/org/ohnlp/medxn/backbone/MedXNDrugBean.java
+++ b/src/main/java/org/ohnlp/medxn/backbone/MedXNDrugBean.java
@@ -1,0 +1,103 @@
+package org.ohnlp.medxn.backbone;
+
+public class MedXNDrugBean {
+    private String code;
+    private String matchedText;
+    private int matchBegin;
+    private int matchEnd;
+    private String matchSentence;
+    private String frequency;
+    private String duration;
+    private String route;
+    private String dosage;
+    private String strength;
+    private String form;
+
+    public String getCode() {
+        return code;
+    }
+
+    public void setCode(String code) {
+        this.code = code;
+    }
+
+    public String getMatchedText() {
+        return matchedText;
+    }
+
+    public void setMatchedText(String matchedText) {
+        this.matchedText = matchedText;
+    }
+
+    public int getMatchBegin() {
+        return matchBegin;
+    }
+
+    public void setMatchBegin(int matchBegin) {
+        this.matchBegin = matchBegin;
+    }
+
+    public int getMatchEnd() {
+        return matchEnd;
+    }
+
+    public void setMatchEnd(int matchEnd) {
+        this.matchEnd = matchEnd;
+    }
+
+    public String getFrequency() {
+        return frequency;
+    }
+
+    public void setFrequency(String frequency) {
+        this.frequency = frequency;
+    }
+
+    public String getDuration() {
+        return duration;
+    }
+
+    public void setDuration(String duration) {
+        this.duration = duration;
+    }
+
+    public String getRoute() {
+        return route;
+    }
+
+    public void setRoute(String route) {
+        this.route = route;
+    }
+
+    public String getDosage() {
+        return dosage;
+    }
+
+    public void setDosage(String dosage) {
+        this.dosage = dosage;
+    }
+
+    public String getStrength() {
+        return strength;
+    }
+
+    public void setStrength(String strength) {
+        this.strength = strength;
+    }
+
+    public String getForm() {
+        return form;
+    }
+
+    public void setForm(String form) {
+        this.form = form;
+    }
+
+    public String getMatchSentence() {
+        return matchSentence;
+    }
+
+    public void setMatchSentence(String matchSentence) {
+        this.matchSentence = matchSentence;
+    }
+}

--- a/src/main/java/org/ohnlp/medxn/backbone/MedXNDrugBean.java
+++ b/src/main/java/org/ohnlp/medxn/backbone/MedXNDrugBean.java
@@ -5,6 +5,8 @@ public class MedXNDrugBean {
     private String matchedText;
     private int matchBegin;
     private int matchEnd;
+    private int matchSection;
+    private String nlpDate;
     private String matchSentence;
     private String frequency;
     private String duration;
@@ -99,5 +101,21 @@ public class MedXNDrugBean {
 
     public void setMatchSentence(String matchSentence) {
         this.matchSentence = matchSentence;
+    }
+
+    public int getMatchSection() {
+        return matchSection;
+    }
+
+    public void setMatchSection(int matchSection) {
+        this.matchSection = matchSection;
+    }
+
+    public String getNlpDate() {
+        return nlpDate;
+    }
+
+    public void setNlpDate(String nlpDate) {
+        this.nlpDate = nlpDate;
     }
 }

--- a/src/main/java/org/ohnlp/medxn/backbone/MedXNOutputToOHDSIFormatTransform.java
+++ b/src/main/java/org/ohnlp/medxn/backbone/MedXNOutputToOHDSIFormatTransform.java
@@ -1,0 +1,97 @@
+package org.ohnlp.medxn.backbone;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.beam.sdk.schemas.Schema;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.Row;
+import org.joda.time.Instant;
+import org.ohnlp.backbone.api.Transform;
+import org.ohnlp.backbone.api.exceptions.ComponentInitializationException;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Transforms MedXN output as produced from {@link org.ohnlp.medxn.backbone.MedXNBackboneTransform} to an OHDSI-compliant format suitable
+ * for the NOTE_NLP table.
+ * <br/>
+ * <b>Important:</b> Requires that the OHDSI vocabulary load query first be run and loaded into backbone resources folder
+ * as ohdsi_rxnorm_map.csv. Please refer to documentation for further details
+ */
+public class MedXNOutputToOHDSIFormatTransform extends Transform {
+    private static ThreadLocal<SimpleDateFormat> sdf = ThreadLocal.withInitial(() -> new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssXXX"));
+
+
+    @Override
+    public void initFromConfig(JsonNode config) throws ComponentInitializationException {
+    }
+
+    @Override
+    public PCollection<Row> expand(PCollection<Row> input) {
+
+        return input.apply(ParDo.of(new DoFn<Row, Row>() {
+            private transient ObjectMapper om;
+            private transient Map<String, Integer> ohdsiConceptMap;
+
+            @Setup
+            public void init() {
+                this.om = new ObjectMapper();
+                this.ohdsiConceptMap = new HashMap<>();
+                try (InputStream resource = MedXNOutputToOHDSIFormatTransform.class.getResourceAsStream("/resources/ohdsi_rxnorm_map.csv")) {
+                    List<String> mappings =
+                            new BufferedReader(new InputStreamReader(resource,
+                                    StandardCharsets.UTF_8)).lines().collect(Collectors.toList());
+                    mappings.forEach(s -> {
+                        String[] args = s.trim().split("\\|");
+                        ohdsiConceptMap.put(args[0], Integer.parseInt(args[1]));
+                    });
+                } catch (Throwable e) {
+                    throw new RuntimeException(e);
+                }
+            }
+
+            @ProcessElement
+            public void processElement(@Element Row input, OutputReceiver<Row> output) throws JsonProcessingException, ParseException {
+                // First transform row schemas
+                List<Schema.Field> fields = new LinkedList<>(input.getSchema().getFields());
+                fields.add(Schema.Field.of("section_concept_id", Schema.FieldType.INT32));
+                fields.add(Schema.Field.of("lexical_variant", Schema.FieldType.STRING));
+                fields.add(Schema.Field.of("snippet", Schema.FieldType.STRING));
+                fields.add(Schema.Field.of("note_nlp_concept_id", Schema.FieldType.INT32));
+                fields.add(Schema.Field.of("note_nlp_source_concept_id", Schema.FieldType.INT32));
+                fields.add(Schema.Field.of("nlp_datetime", Schema.FieldType.DATETIME));
+                fields.add(Schema.Field.of("term_modifiers", Schema.FieldType.STRING));
+                Schema schema = Schema.of(fields.toArray(new Schema.Field[0]));
+
+                MedXNDrugBean bean = om.readValue(input.getString("nlp_output_json"), MedXNDrugBean.class);
+
+                // Now generate an output row
+                Row out = Row.withSchema(schema)
+                        .addValues(input.getValues())
+                        .addValue(0)
+                        .addValue(bean.getMatchedText())
+                        .addValue(bean.getMatchSentence())
+                        .addValue(ohdsiConceptMap.getOrDefault(bean.getCode(), 0))
+                        .addValue(0)
+                        .addValue(new Instant(sdf.get().parse(bean.getNlpDate()).getTime()))
+                        .addValue(om.writeValueAsString(bean))
+                        .build();
+                output.output(out);
+            }
+        }));
+    }
+
+}

--- a/src/main/java/org/ohnlp/medxn/backbone/README.md
+++ b/src/main/java/org/ohnlp/medxn/backbone/README.md
@@ -24,7 +24,7 @@ An example backbone component configuration follows:
 }
 ```
 
-### MedXNOutputToOHDSIFormatTransform
+#### MedXNOutputToOHDSIFormatTransform
 Transforms the results from MedXNBackboneTransform into a format compliant with OHDSI's NOTE_NLP table. Note that it is 
 assumed that the note_id will also be supplied as part of the input rows
 

--- a/src/main/java/org/ohnlp/medxn/backbone/README.md
+++ b/src/main/java/org/ohnlp/medxn/backbone/README.md
@@ -1,0 +1,51 @@
+## Package Summary
+This package contains necessary classes to run medxn as part of a [Backbone](https://www.github.com/OHNLP/Backbone)
+pipeline. 
+
+### Component Descriptions
+#### MedXNBackboneTransform
+A Base Backbone Component for Running MedXN on Input Text based on Configuration Input Column
+
+Takes the input row, duplicates all values, and adds a `nlp_output_json` column for each 
+extracted drug mention as output. `nlp_output_json` will be a jackson-serialized instance of
+`MedXNDrugBean` and can be converted back to Java object format via the following code snippet:
+```java
+ObjectMapper om = new ObjectMapper();
+MedXNDrugBean pojoValue = om.readValue(row.getString("nlp_output_json"), MedXNDrugBean.class);
+```
+
+An example backbone component configuration follows:
+```json
+{
+      "clazz": "org.ohnlp.medxn.backbone.MedXNBackboneTransform",
+      "config": {
+        "input": "note_text"
+      }
+}
+```
+
+### MedXNOutputToOHDSIFormatTransform
+Transforms the results from MedXNBackboneTransform into a format compliant with OHDSI's NOTE_NLP table. Note that it is 
+assumed that the note_id will also be supplied as part of the input rows
+
+An example backbone component configuration follows:
+```json
+{
+      "clazz": "org.ohnlp.medxn.backbone.MedXNOutputToOHDSIFormatTransform",
+      "config": {}
+}
+```
+
+### Installation Instructions
+1. Download the latest MedXN Distribution (or use the generated dist from ant build) and copy 
+the generated MedXN JAR file to the backbone installation's `modules` folder
+2. If using the MedXNOutputToOHDSIFormatTransform transformation, retrieve the RxNorm->OHDSI mapping
+by connecting to your Athena/OHDSI database instance and saving the results of the below query 
+(without column headers) to `your_backbone_install/resources/ohdsi_rxnorm_map.csv`
+```sql
+SELECT concept_code, concept_id FROM your_cdm_schema.concept WHERE vocabulary_id IN (LIST_OF_OHDSI_VOCAB_IDS_TO_INCLUDE)
+```
+For MedXN's Purposes, We suggest RxNorm and RxNorm Extensions for inclusion in LIST_OF_OHDSI_VOCAB_IDS_TO_INCLUDE
+3. Make appropriate changes to your pipeline configuration to include the MedXN Components to run
+4. Package everything using backbone's distributed `package_modules_and_configs.{bat|sh}`.
+

--- a/src/main/resources/NOTICE
+++ b/src/main/resources/NOTICE
@@ -1,0 +1,5 @@
+RxNorm
+======
+This product uses publicly available data courtesy of the U.S. National Library of Medicine (NLM), National Institutes
+of Health, Department of Health and Human Services; NLM is not responsible for the product and does not endorse or
+recommend this or any other product.

